substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/NativeImageTracingAgent.java

```diff
@@ -644,10 +605,10 @@ public static void onUnload(@SuppressWarnings("unused") JNIJavaVM vm) {
         }
 
         /*
-         * Agent shutdown is tricky: apparently we can still have events at the same time as this
-         * function executes, so we would need to synchronize. We could do this with a combined
-         * shared+exclusive lock, but that adds some cost to all events. We choose to leak a few
-         * handles and some memory for now -- this agent isn't supposed to be attached only
+         * JvmtiAgentBase shutdown is tricky: apparently we can still have events at the same time
```

Comment:
That's more about JVMTI agents in general. Leaving `Agent shutdown [...]` or using `JVMTI agent shutdown [...]` would be better.

Reply:
I have changed it back to `Agent shutdown [...]`.